### PR TITLE
adding buy options block

### LINKED ISSUE #61: [Product Pages] Clickable logos to external sites
For site pages where we have clickable logos which redirect to external sites (links are different for different products) for product purchase:
e.g https://www.fernleaf.com.my/en/products/milk-powder-for-whole-family.html
https://www.fernleaf.com.my/en/products/formulated-milk-powder-for-children.html

Screenshot 2023-08-02 at 1 06 59 PM

Approach: author these in the product pages itself (add SVGs? to an icons folder in sharepoint, preview/publish them). Then just reference the icon using :icon-name: and link that in word.

### PR DESCRIPTION
**Pull request by mkbansal1**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #61 

Test URLs:
- Before: https://main--fonterra-fernleaf--hlxsites.hlx.page/en/products/uht-milk
- After: https://61-buy-options--fonterra-fernleaf--hlxsites.hlx.live/en/products/uht-milk

### CODE REVIEW

**Comment by ujjwal5** [comment]
@mkbansal1 As discussed, we can reuse existing fetchplaceholders() from lib-franklin.js and pass getLanguage() as argument.
An example: https://github.com/hlxsites/fonterra-fernleaf/blob/main/scripts/modals/languages.js#L66

**Comment by mkbansal1** [comment]
> @mkbansal1 As discussed, we can reuse existing fetchplaceholders() from lib-franklin.js and pass getLanguage() as argument. An example: https://github.com/hlxsites/fonterra-fernleaf/blob/main/scripts/modals/languages.js#L66

Agree, i have reverted my logic

File: blocks/buy-options/buy-options.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+let buyOptions = [];
+
+async function preProcess(block, placeholder) {
+  buyOptions = [];
+  [...block.children].forEach((row) => {
+    const data = {};
+    data.brand = row.children[0].innerText;
+    data.link = row.children[1].innerText;
+    data.img = placeholder[`product${data.brand}`];
```
Comment (ujjwal5) [inline blocks/buy-options/buy-options.js:18]: @mkbansal1 any specific reason why we have kept company images language specific?

File: blocks/buy-options/buy-options.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                <img alt="${row.brand}" src="${row.img}" width="100" height="42">
+            </a>`;
+
+let buyOptions = [];
+
+async function preProcess(block, placeholder) {
+  buyOptions = [];
+  [...block.children].forEach((row) => {
+    const data = {};
+    data.brand = row.children[0].innerText;
```
Comment (ParthasharathiB) [inline blocks/buy-options/buy-options.js:16]: Include a null check on the children object

Comment (mkbansal1) [reply blocks/buy-options/buy-options.js]: Done

File: blocks/buy-options/buy-options.js (hunk 1 above)
Comment (ujjwal5) [reply blocks/buy-options/buy-options.js]: Do we have a use case where we have different company image for different language. I think company logo/images is consistent across languages.
Using placeholder for this use case would be an overkill. Plus an author will have to maintain it at 2 different places(more chances of error). We can simply search/use ${data.brand} named image from images folder here.
WDYT ?

**Review by sselvara — CHANGES_REQUESTED** [review]

File: blocks/buy-options/buy-options.js (hunk 1 above)
Comment (mkbansal1) [reply blocks/buy-options/buy-options.js]: We are using placeholders to configure image paths (brand images, splash images ...), cant use sharefolder named based image as it does 301 redirect to media bus url.
Agree that it does not make sense to manage such images at two places, i have created a placeholder file at root level for such configurations.

File: blocks/buy-options/buy-options.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { getLanguage } from '../../scripts/scripts.js';
+
+const getHTML = (row) => `<a target="_blank" href="${row.link}" title="${row.brand}" aria-label="${row.brand}">
+                <img alt="${row.brand}" src="${row.img}" width="100" height="42">
+            </a>`;
+
+let buyOptions = [];
+
+async function preProcess(block, placeholder) {
+  buyOptions = [];
```
[the remark this answers is not in the document]
Comment (mkbansal1) [reply blocks/buy-options/buy-options.js]: Done

File: blocks/buy-options/buy-options.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.section div.buy-options-wrapper,
+.buy-options-container div:first-child {
+  padding: 0;
+}
+
```
[the remark this answers is not in the document]
Comment (mkbansal1) [reply blocks/buy-options/buy-options.css:5]: Done

File: blocks/buy-options/buy-options.js (hunk 1 above)
Comment (ujjwal5) [reply blocks/buy-options/buy-options.js]: Should we remove these entries from placeholder files under /en and /ms ?

Comment (mkbansal1) [reply blocks/buy-options/buy-options.js]: Already done

Comment (ujjwal5) [reply blocks/buy-options/buy-options.js]: Thanks I see we have removed it now.

**Review by ujjwal5 — APPROVED** [review]

**Review by sselvara — APPROVED** [review]

**Pull request merged**